plugin/crates.vim

```diff
@@ -43,15 +43,31 @@ function! s:job_callback_nvim_stdout(_job_id, data, _event) dict abort
   call extend(self.stdoutbuf, a:data[1:])
 endfunction
 
+function! s:job_callback_vim_stdout(channel, message) dict abort
+  let self.stdoutbuf[0] .= a:message
+endfunction
+
 function! s:job_callback_nvim_exit(_job_id, exitval, _event) dict abort
   call self.callback(a:exitval)
 endfunction
 
+function! s:job_callback_vim_exit(_channel, exitval) dict abort
+  let self.exitval = a:exitval
+endfunction
+
+function! s:job_callback_vim_close(_channel) dict abort
+  call self.callback(self.exitval)
+endfunction
+
 function! s:callback_show_latest_version(exitval) dict abort
   if a:exitval
     echomsg "D'oh! Got ". a:exitval
     return
   endif
+  if self.stdoutbuf[0] == ''
+    echomsg "D'oh! Got no output for crate " . self.crate
+    return
+  endif
```

Comment:
This is technically unnecessary, but I found it useful while debugging some issues -- and it might be a more useful error than "couldn't parse json"